src/js/main.js

```diff
@@ -50,64 +58,65 @@ if (navigator && navigator.clipboard && navigator.permissions) {
 	});
 }
 
-// When someone navigates directly to a checklist item using its "Share Link" 
-// like a11yproject.com/checklist/#validate-your-html, the item with the
-// matching id attribute will be scrolled into view. Then, if JS is enabled,
-// this code will open its associated <details> element. 
-function openLinkedCheckListItem() {			
-	var checklistItems = document.querySelectorAll("[data-checklist-item-id]");
-
-	checklistItems.forEach(function (item) {
-		var shareLink = "#" + item.getAttribute("data-checklist-item-id");
-		var isItemLinked = shareLink === document.location.hash;
-
-		if (isItemLinked) {
-			item.setAttribute("open", true);
-		}
-	});
+/**
+ * If someone has opened the `checklist` page using one a checklist "Share link"
+ * (ex: a11yproject.com/checklist/#validate-your-html) the item with the
+ * corresponding id will scroll into view. Then, if JS is enabled,
+ * this function will open its associated <details> element.
+ */
+function openLinkedCheckListItem() {
+	var hash = window.location.hash.substr(1);
+	var checklistItem = document.querySelector(
+		"[data-checklist-item-id=" + hash + "]"
+	);
+
+	if (checklistItem) {
+		checklistItem.setAttribute("open", true);
+	}
 }
 // Store checklist status ---------------------------------------------------
-function storeChecklistItems(item) {
-	localStorage.setItem(item, 'checked');
+function storeChecklistItem(checkboxId) {
+	localStorage.setItem(checkboxId, 'checked');
 }
 
-function removeChecklistItems(item) {
-	localStorage.removeItem(item);
+function removeChecklistItem(checkboxId) {
+	localStorage.removeItem(checkboxId);
 }
 
-function getChecklistItemVal(item) {
-	return localStorage.getItem(item);
-}
+function processChecklistClick(checkboxSelector) {
+	document.addEventListener("change", function(event) {
+		var target = event.target;
 
-function checkChecklistExists(item) {
-	return getChecklistItemVal(item) === 'checked';
-}
+		if (!target.matches(checkboxSelector)) {
+			return;
+		}
 
-function processChecklistClick(items) {
-	items.forEach(function(element){
-		element.addEventListener('change', function(event) {
-			if (checkChecklistExists(event.target.id)) {
-				removeChecklistItems(event.target.id);
-			} else {
-				storeChecklistItems(event.target.id);
-			}
-		});
+		if (target.checked) {
+			storeChecklistItem(target.id);
+		} else {
+			removeChecklistItem(target.id);
+		}
 	});
 }
 
-function renderLocalStorageItems(items) {
-	items.forEach(function(element){
-		if (checkChecklistExists(element.id)) {
-			element.checked = true;
-		}
-	});
+function populateChecklistFromLocalStorage(checkboxSelector) {
+	var items = document.querySelectorAll(checkboxSelector);
+	var length = items.length;
+	for (var i = 0; i < length; ++i) {
```

Comment:
Any particular reason for replacing the forEach with a for loop?

Reply:
It's probably a micro-optimization, but: iterating with `forEach` is always more expensive than a `for...` loop, because `forEach` requires function calls; for loops do not.